### PR TITLE
Update/sw get tweaks

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

* Adds more help description
* uses positional argument to filter component
* Updates query to include appId (needed for correct env naming convention)
* Switches to manual formatter

## Steps to Test

```
npm run build && ./dist/bin/vip-config-software-get.js @4092 nodejs
```

### CODE REVIEW

File: src/bin/vip-config-software-get.js [hunk 1]
```diff
@@ -16,26 +16,54 @@ import chalk from 'chalk';
 import command from 'lib/cli/command';
 import { appQuery, appQueryFragments } from 'lib/config/software';
 
+// Command examples
+const examples = [
+	{
+		usage: 'vip config software get wordpress',
+		description: 'Read current software settings for wordpress',
```
Comment (rinatkhaziev) [inline src/bin/vip-config-software-get.js]:
```suggestion
		description: 'Read current software settings for WordPress',
```

File: src/bin/vip-config-software-get.js [hunk 2]
```diff
@@ -16,26 +16,54 @@ import chalk from 'chalk';
...
-		softwareSettings.php,
-		softwareSettings.muplugins,
-		softwareSettings.nodejs,
-	]
+	let choosenSettings = [];
+	if ( arg.length > 0 ) {
+		const component = arg[ 0 ];
+		if ( ! softwareSettings[ component ] ) {
+			// TODO throw user error
+			console.log( chalk.yellow( 'Note:' ), `Software settings for ${ component } are not supported for this environmnet.` );
```
Comment (rinatkhaziev) [inline src/bin/vip-config-software-get.js]:
```suggestion
			console.log( chalk.yellow( 'Note:' ), `Software settings for ${ component } are not supported for this environment.` );
```

File: src/bin/vip-config-software-get.js [hunk 3]
```diff
@@ -16,26 +16,54 @@ import chalk from 'chalk';
...
 		process.exit();
 	}
 
-	const preFormated = [
-		softwareSettings.wordpress,
-		softwareSettings.php,
-		softwareSettings.muplugins,
-		softwareSettings.nodejs,
-	]
+	let choosenSettings = [];
```
Comment (rinatkhaziev) [inline src/bin/vip-config-software-get.js]: Typo here and on, should probably read `chosenSettings` not `choosen`

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
A few typos here and there. 

Should we also include the `--format` options in examples?

**Comment by pschoffer** [comment]
I added the `--format` example and fixed a typo. - Thanks @rinatkhaziev !

Also, the formatting logic was changed based on @chriszarate 's suggestion - https://github.com/Automattic/vip/pull/1083

File: src/bin/vip-config-software-get.js [hunk 4]
```diff
@@ -14,28 +14,57 @@ import chalk from 'chalk';
...
 	format: true,
-} ).argv( process.argv, async ( arg: string[], { env } ) => {
-	const { softwareSettings } = env;
+	usage: 'vip config software get <wordpress|php|nodejs|muplugins>',
+} ).examples( examples ).argv( process.argv, async ( arg: string[], opt ) => {
+	const { softwareSettings } = opt.env;
 
 	if ( softwareSettings === null ) {
+		// TODO throw user error
 		console.log( chalk.yellow( 'Note:' ), 'Software settings are not supported for this environmnet.' );
```
Comment (chriszarate) [inline src/bin/vip-config-software-get.js]:
```suggestion
		console.log( chalk.yellow( 'Note:' ), 'Software settings are not supported for this environment.' );
```

File: src/bin/vip-config-software-get.js [hunk 5]
```diff
@@ -14,28 +14,57 @@ import chalk from 'chalk';
...
+	wildcardCommand: true,
 	format: true,
-} ).argv( process.argv, async ( arg: string[], { env } ) => {
-	const { softwareSettings } = env;
+	usage: 'vip config software get <wordpress|php|nodejs|muplugins>',
+} ).examples( examples ).argv( process.argv, async ( arg: string[], opt ) => {
+	const { softwareSettings } = opt.env;
 
 	if ( softwareSettings === null ) {
+		// TODO throw user error
```
Comment (chriszarate) [inline src/bin/vip-config-software-get.js]: what does this mean?

File: src/bin/vip-config-software-get.js [hunk 6]
```diff
@@ -14,28 +14,57 @@ import chalk from 'chalk';
...
+	} else {
+		chosenSettings = [
+			softwareSettings.wordpress,
+			softwareSettings.php,
+			softwareSettings.muplugins,
+			softwareSettings.nodejs,
+		];
+	}
+
+	const preFormated = chosenSettings
```
Comment (chriszarate) [inline src/bin/vip-config-software-get.js]:
```suggestion
	const preFormatted = chosenSettings
```

File: src/bin/vip-config-software-get.js [hunk 7]
```diff
@@ -50,5 +79,5 @@ command( {
 			};
 		} );
 
-	return preFormated;
+	console.log( formatData( preFormated, opt.format ) );
```
Comment (chriszarate) [inline src/bin/vip-config-software-get.js]:
```suggestion
	console.log( formatData( preFormatted, opt.format ) );
```

File: src/bin/vip-config-software-get.js (hunk 5 above)
Comment (pschoffer) [reply src/bin/vip-config-software-get.js]: Oh right! Thanks for noticing the TODOs. I introduced these UserError that don't print the usual "Contact customer support" error, but show the error to the user, while still using regular error processing.

Now it got merged so I can use it here. I updated the code and removed the TODOs.

**Review by chriszarate — APPROVED** [review]
👍 🎉

**Pull request merged**